lib/Db/RegistrationMapper.php

```diff
@@ -84,16 +84,16 @@ public function insert(Entity $entity) {
 	/**
 	 * @param Registration $registration
 	 */
-	public function generateNewToken(Registration &$registration) {
+	public function generateNewToken(Registration $registration) {
```

Comment:
Bonus points if you return the object so you can chain it.

Reply:
not for now